_studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp

```diff
@@ -4321,34 +4317,21 @@ mfxStatus MfxHwH264Encode::CheckVideoParamQueryLike(
         }
     }
 
-#if MFX_VERSION > 1021
+
     if (extRoi->NumROI && extRoi->ROIMode != MFX_ROI_MODE_QP_DELTA && extRoi->ROIMode != MFX_ROI_MODE_PRIORITY)
     {
         unsupported = true;
         extRoi->NumROI = 0;
     }
 
-    if (extRoi->NumROI && par.mfx.RateControlMethod != MFX_RATECONTROL_CQP &&
-        extRoi->ROIMode == MFX_ROI_MODE_QP_DELTA && hwCaps.ddi_caps.ROIBRCDeltaQPLevelSupport == 0)
+    if ((extRoi->NumROI && (extRoi->ROIMode == MFX_ROI_MODE_QP_DELTA ||
+        extRoi->ROIMode == MFX_ROI_MODE_PRIORITY ))&&
+        hwCaps.ddi_caps.ROIBRCDeltaQPLevelSupport == 0)
```

Comment:
Senseless over-complication, ROIMode value already verified above.
This check is enough now:
if (extRoi->NumROI && hwCaps.ddi_caps.ROIBRCDeltaQPLevelSupport == 0)

Reply:
done